modular_citadel/code/datums/status_effects/chems.dm

```diff
@@ -0,0 +1,704 @@
+			to_chat(o, "<span class='notice'>Your bountiful bosom is so rich with mass, you seriously doubt you'll be able to fit any clothes over it.</b></span>")
+		return ..()
+
+/datum/status_effect/chem/BElarger/tick(mob/living/carbon/human/H)//If you try to wear clothes, you fail. Slows you down if you're comically huge
+	var/mob/living/carbon/human/o = owner
+	var/obj/item/organ/genital/breasts/B = o.getorganslot("breasts")
+	moveCalc = 1+((round(B.cached_size) - 9)/3) //Afffects how fast you move, and how often you can click.
+	if(!B)
+		o.remove_movespeed_modifier("megamilk")
+		sizeMoveMod(1)
+		owner.remove_status_effect(src)
+	var/items = o.get_contents()
+	for(var/obj/item/W in items)
+		if(W == o.w_uniform || W == o.wear_suit)
+			o.dropItemToGround(W, TRUE)
+			playsound(o.loc, 'sound/items/poster_ripped.ogg', 50, 1)
+			to_chat(owner, "<span class='warning'>Your enormous breasts are way too large to fit anything over them!</b></span>")
+	if (B.size == "huge")
+		if(prob(1))
+			to_chat(owner, "<span class='notice'>Your back is feeling a little sore.</span>")
+			var/target = o.get_bodypart(BODY_ZONE_CHEST)
+			o.apply_damage(0.1, BRUTE, target)
+		if(!B.cached_size == B.breast_values[B.prev_size])
+			o.add_movespeed_modifier("megamilk", TRUE, 100, NONE, override = TRUE, multiplicative_slowdown = moveCalc)
+			sizeMoveMod(moveCalc)
+		return ..()
+	else if (B.breast_values[B.size] > B.breast_values[B.prev_size])
+		o.add_movespeed_modifier("megamilk", TRUE, 100, NONE, override = TRUE, multiplicative_slowdown = moveCalc)
+		sizeMoveMod(moveCalc)
+	else if (B.breast_values[B.size] < B.breast_values[B.prev_size])
+		o.add_movespeed_modifier("megamilk", TRUE, 100, NONE, override = TRUE, multiplicative_slowdown = moveCalc)
+		sizeMoveMod(moveCalc)
+	if((B.cached_size) < 16)
+		switch(round(B.cached_size))
+			if(9)
+				if (GLOB.breasts_size_list[B.prev_size] != GLOB.breasts_size_list[B.size])
+					to_chat(o, "<span class='notice'>Your expansive chest has become a more managable size, liberating your movements.</b></span>")
+			if(10 to INFINITY)
+				if (GLOB.breasts_size_list[B.prev_size] != GLOB.breasts_size_list[B.size])
+					to_chat(H, "<span class='warning'>Your indulgent busom is so substantial, it's affecting your movements!</b></span>")
+		if(prob(2))
+			to_chat(owner, "<span class='notice'>Your back is feeling a little sore.</span>")
+		..()
+
+/datum/status_effect/chem/BElarger/on_remove(mob/living/carbon/M)
+	owner.remove_movespeed_modifier("megamilk")
+	sizeMoveMod(1)
+
+/datum/status_effect/chem/BElarger/proc/sizeMoveMod(var/value)
+	if(cachedmoveCalc == value)
+		return
+	owner.next_move_modifier /= cachedmoveCalc
+	owner.next_move_modifier *= value
+	cachedmoveCalc = value
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/status_effect/chem/PElarger
```

Comment:
please have typepaths be descriptive : "pelarger" doesn't tell us much, "penis_enlarger" does. also, avoid uppercase in typepaths whenever possible.

Reply:
I honestly try to avoid writing that word as much as possible.